--- a/setup_libuv.py.orig	2017-07-08 16:12:17.000000000 +0300
+++ a/setup_libuv.py	2020-03-29 15:38:55.768185295 +0300
@@ -143,7 +143,6 @@
 
         if sys.platform.startswith('linux'):
             self.compiler.add_library('dl')
-            self.compiler.add_library('rt')
         elif sys.platform == 'darwin':
             self.compiler.define_macro('_DARWIN_USE_64_BIT_INODE', 1)
             self.compiler.define_macro('_DARWIN_UNLIMITED_SELECT', 1)
